### PR TITLE
Add flip function

### PR DESCRIPTION
**Pull request by ronekko**
This PR adds `F.flip(x, axis)`. (Chainer has `F.flipud`, `F.fliplr` already).
The implementation is based on `FunctionNode`.

### CODE REVIEW

**Comment by ronekko** [comment]
Since `numpy.flip` is supported from version 1.12.0, `_flip` is its alternative for older version.

File: chainer/functions/array/flip.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            return xp.flip(inputs[0], self.axis),
+        else:
+            return _flip(inputs[0], self.axis),
+
+    def backward(self, indexes, grad_outputs):
+        return flip(grad_outputs[0], self.axis),
+
+
+def flip(x, axis):
+    """Flip an input variable in reverse order along the given axis.
```
Comment (unnonouno) [inline chainer/functions/array/flip.py]: "Flip" -> "Flips"

File: chainer/functions/array/flip.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def _flip(array, axis):
+    indices = [slice(None)] * array.ndim
+    indices[axis] = slice(None, None, -1)
+    return array[indices]
+
+
+class Flip(function_node.FunctionNode):
+    """Flip an input variable in reverse order along the given axis."""
```
Comment (unnonouno) [inline chainer/functions/array/flip.py]: Append "s"
"Flip" -> "Flips"

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by ronekko** [comment]
Thank you for the review.
I fixed the docstring.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM!! Thank you

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
Thank you very much!!

**Pull request merged**